### PR TITLE
Move SHMEM_CTX_DEFAULT to new library handles section

### PR DESCRIPTION
**Pull request by jdinan**
Proposed change to make SHMEM_CTX_DEFAULT a link-time constant.

Please don't merge until this clears committee discussion at the 11/9 meeting.

PDF Preview: [main_spec.pdf](https://github.com/RaymondMichael/specification/files/1451875/main_spec.pdf)

### CODE REVIEW

**Comment by jdinan** [comment]
Note that the text describing the handle semantics is lifted from MPI 3.1.  I'm open to suggestions on how to improve this, but it seemed like a reasonable place to start.

**Comment by jdinan** [comment]
Closes https://github.com/openshmem-org/specification/issues/178

**Comment by jdinan** [comment]
@RaymondMichael This PR is ready for your section committee to process and merge.

File: content/library_handles.tex [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The \openshmem library provides a set of predefined named constant handles.
+All named constants can be used in initialization expressions or assignments,
+but not necessarily in array declarations or as labels in C switch statements.
```
Comment (nspark) [inline content/library_handles.tex]: `C` -> `\Cstd`

**Comment by nspark** [comment]
@jdinan Do we really want to call these "predefined named _constant_ handles" (emphasis mine)? [You even note](https://github.com/openshmem-org/specification/issues/178#issuecomment-341739671) that these are not necessarily identifiers for `const`-specified objects.

I think "predefined named handle" should be sufficient. If any user program modifies `SHMEM_CTX_DEFAULT`, the behavior should be undefined.

**Comment by jdinan** [comment]
@nspark I had a similar thought when I was drafting this.  Yes, I think "constant" is needed to guarantee that a user can capture the value of this handle in a variable and use it without concern that the value of the handle could change.

File: content/library_handles.tex (hunk 1 above)
Comment (jdinan) [reply content/library_handles.tex]: PR Posted: https://github.com/RaymondMichael/specification/pull/19

**Comment by BryantLam** [comment]
Is value capture of the handle only constant if the library has been initialized? This is the first instance that I can recall that a const variable initialization will depend on the library being initialized.

From `shmem_init` (emphasis added):
> shmem_init allocates and initializes **resources** used by the OpenSHMEM library. It is a collective operation that all PEs must call before any other OpenSHMEM **routine** may be called.

The Library Handles intro does not mention the word "resource" in it.

**Comment by jdinan** [comment]
The value of the constant must be available at link time.  This comes
before initialization.

**Pull request merged**